stacklight_tests/clients/influxdb_grafana_api.py

```diff
@@ -161,113 +198,325 @@ def get_all_measurements(self):
 
     def get_panel_queries(self):
         panel_queries = {}
-        identifier = 0
-        for row in self.dash_dict["dashboard"]["rows"]:
-            for panel in row["panels"]:
-                panel_name = "{}: {}".format(row["title"], panel["title"])
-                for target in panel.get("targets", [{}]):
-                    query = target.get("query")
-                    if query:
-                        query_name = "{}: {}: {}".format(
-                            identifier, panel_name, target.get(
-                                "measurement",
-                                self._parse_measurement_from_query(query)))
-                        identifier += 1
-                        assert query_name not in panel_queries, query_name
-                        panel_queries[query_name] = (
-                            (query,
-                             self._compile_query(query, self.templates)))
+        for panel, row in self.panels:
+            panel_name = "{}->{}".format(row["title"], panel["title"] or "n/a")
+            for target in panel.get("targets", []):
+                query = self.build_query(target)
+                table = target.get(
+                    "measurement", self._influxdb_api.parse_measurement(query))
+                query_name = "{}:{}->{}->RefId:{}".format(
+                    panel["id"], panel_name, table, target.get("refId", "A"))
+                panel_queries[query_name] = query, table
         return panel_queries
 
-    def classify_all_dashboard_queries(self):
-        ok_queries = {}
-        failed_queries = {}
-        no_measurements_queries = {}
-        for key, (raw_query, query) in self.get_panel_queries().items():
+    def classify_query(self, raw_query, table):
+        if table and (table not in self.available_measurements):
+            return "no_table", raw_query
+        results = collections.defaultdict(list)
+        possible_templates = self.get_all_templates_for_query(raw_query)
+        for template in possible_templates:
+            query = self._influxdb_api.compile_query(raw_query, template)
+            raw_result = self._influxdb_api.do_influxdb_query(query).json()
             try:
-                if "ceph" in query:
-                    # NOTE(rpromyshlennikov): ceph is disabled in most cases
-                    continue
-                query_table = self._parse_measurement_from_query(query)
-                if query_table and (
-                        query_table not in self.available_measurements):
-                    no_measurements_queries[key] = raw_query, query, {}
-                    continue
-                raw_result = self._influxdb_api.do_influxdb_query(query).json()
                 result = raw_result["results"][0]
                 assert result["series"][0]["values"]
-                ok_queries[key] = raw_query, query, result
+                results["ok"].append((template, result))
```

Comment:
It was done like this because I couldn't decide which message to show on test fail (I have tried several approaches). Now it's redundant, so, I have removed it.